proto_parser/src/proto_parser.cc

```diff
@@ -0,0 +1,114 @@
+/*
+*   Copyright 2020 Google LLC
+*
+*   Licensed under the Apache License, Version 2.0 (the "License");
+*   you may not use this file except in compliance with the License.
+*   You may obtain a copy of the License at
+*
+*    https://www.apache.org/licenses/LICENSE-2.0
+*
+*   Unless required by applicable law or agreed to in writing, software
+*   distributed under the License is distributed on an "AS IS" BASIS,
+*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*   See the License for the specific language governing permissions and
+*   limitations under the License.
+*/
+
+#include "proto_parser.h"
+#include "message_value.h"
+#include "primitive_value.h"
+
+namespace wireless_android_play_analytics {
+
+void ProtoParser::PopulateFields(int& prev_field_line, 
+    const google::protobuf::TextFormat::ParseInfoTree& tree,
+    const google::protobuf::Message& message,
+    ProtoValue* proto_value, int indent_count) {
+  //TODO(alexanderlin): Add Unit test.
+  const google::protobuf::Reflection* reflection = message.GetReflection();
+  const google::protobuf::Descriptor* descriptor = message.GetDescriptor();
+  MessageValue* message_tmp = 
+      dynamic_cast<MessageValue*>(proto_value);
+  assert(message_tmp);
+  std::vector<std::unique_ptr<ProtoValue>>& message_field = message_tmp->
+      GetFieldsMutable();
+  int field_count = descriptor->field_count();
+  std::vector<FieldInfo> field_info_list;
+  // Get the order of textproto
+  for (int i = 0; i < field_count; ++i) {
+    const google::protobuf::FieldDescriptor* field_descriptor = 
+        descriptor->field(i);
+    if (field_descriptor->is_repeated()) {
+      int field_size = reflection->FieldSize(message, field_descriptor);
+      for (int j = 0; j < field_size; ++j) {
+        field_info_list.push_back(FieldInfo(GetLocation(tree, field_descriptor, 
```

Comment:
field_info_list.push_back(FieldInfo(/*line=*/GetLocation(tree, field_descriptor, 
            j), /*index=*/j, field_descriptor));

For better readability.